### PR TITLE
zephyr: boot: added support for nrf52840_pca10056 board

### PR DESCRIPTION
**Pull request by ChristopherBittner**
Added configuration file for nrf52840_pca10056.
Fixed UART name required for pca10056.

Signed-off-by: Chris Bittner <chris.bittner@nordicsemi.no>

### CODE REVIEW

**Comment by ChristopherBittner** [comment]
cc @nvlsianpu

**Comment by lemrey** [comment]
Hi @ChristopherBittner , why are you changing the default configuration to enable Serial recovery?

**Comment by ChristopherBittner** [comment]
Hi @lemrey 
I've built this configuration based on the nrf52840_pca10059.conf where serial recovery is already enabled. So I just followed what we had there but to be used with the development kit.

File: boot/zephyr/boards/nrf52840_pca10056.conf [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# The UART is used for Serial Recovery, so logging requires
```
Comment (nvlsianpu) [inline boot/zephyr/boards/nrf52840_pca10056.conf]: Please remove this file. We use UART serial recovery, no-multithreading for smaller footprint which is configured well via kconfig file.

Comment (ChristopherBittner) [reply boot/zephyr/boards/nrf52840_pca10056.conf]: OK. 
Ill close the PR then. No point in merging without that file.

File: boot/zephyr/serial_adapter.c [hunk 2]
```diff
@@ -191,7 +191,7 @@ static int
 boot_uart_fifo_init(void)
 {
 #ifdef CONFIG_BOOT_SERIAL_UART
-	uart_dev = device_get_binding(CONFIG_UART_CONSOLE_ON_DEV_NAME);
+	uart_dev = device_get_binding(DT_UART_CONSOLE_ON_DEV_NAME);
```
Comment (lemrey) [inline boot/zephyr/serial_adapter.c]: @ChristopherBittner @nvlsianpu This is probably still needed regardless.

Comment (nvlsianpu) [reply boot/zephyr/serial_adapter.c]: I opened PR

**Pull request closed without merging**